examples/xgboost/README.md

```diff
@@ -0,0 +1,164 @@
+
+In addition to basic uniform shrinkage setting where all clients have the same learning rate, based on our research, we enabled scaled shrinkage across clients for weighted aggregation according to each client's data size.
+
+
+## Run automated experiments
+### Prepare training configs with data splits
+First, we prepare the configs for several training data splits and scenarios by running
+```
+bash train_config_gen.sh
+```
+To be specific, this script calls the python script `./utils/prepare_train_config.py`. The arguments are:
+- site_num: total number of sites
+- nthread: nthread parameter for xgboost, this is set according to system resource 
+- size_total: total number of instances, for HIGGS dataset it is 11 million
+- size_valid: validation size, for the experiments here, it is 1 million, indicating the first 1 million instances will be used as standalone validation set. 
+- split_method: how to split the dataset, can be uniform, linear, square, and exponential
+- lr_mode: whether to use uniform or scaled shrinkage
+This will generate data splits for two client sizes: 5 and 20, and 3 split conditions: uniform, square, and exponential. Users can further customize it for more experiments.
+> **_NOTE:_** The generated train config files will be stored in the folder `./train_configs`, and will be used by job_configs by specifying it with `config_fed_client.json` 
+
+
+### Prepare job configs under various training schemes
+We then prepare the job configs for NVFlare jobs corresponding to various settings by running
+```
+bash job_config_gen.sh
+```
+To be specific, this script calls the python script `./utils/prepare_job_config.py`. It modifies settings from a base config `./job_configs/higgs_base`, and relies on the train_configs generated in the last step.
+
+Here, we generated in total 10 different configs: five for each of the 5/20-client settings:
+- bagging training with uniform data split and uniform shrinkage 
+- bagging training with non-uniform data split and uniform shrinkage 
+- bagging training with non-uniform data split and scaled shrinkage
+- cyclic training with uniform data split 
+- cyclic training with non-uniform data split 
+
+Cyclic training always use uniform shrinkage 
+> **_NOTE:_** The generated job config files will be stored in the folder `./job_configs`
+
+### Start the FL system and submit jobs
+Next, we will start the FL system and submit jobs to start FL training automatically.
+
+Start the FL system with either 5 clients or 20 clients
+```
+bash start_fl_poc.sh 5
+```
+or
+```
+bash start_fl_poc.sh 20
+```
+This script will start the FL server and clients automatically to run FL experiments on localhost. 
```

Comment:
before users start they need to "create poc workspace"?

Reply:
good catch! added to readme below "Set up FL workspace"